src/Symfony/Component/Console/CommandsResolver/CommandResolver.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Console\CommandsResolver;
+
+use Symfony\Component\Console\Command\Command;
+
+/**
+ * @author Ivan Shcherbak <dev@funivan.com>
+ */
+class CommandResolver implements CommandResolverInterface
+{
+    private $commands = array();
+
+  /**
+   * {@inheritdoc}
+   */
+  public function add(Command $command)
+  {
+      $this->commands[$command->getName()] = $command;
+
+      foreach ($command->getAliases() as $alias) {
+          $this->commands[$alias] = $command;
+      }
+
+      return $this;
```

Comment:
IMO, you should not return anything. Returning the command object does not make sense: if the user need it, they can save it first as they are passing it as argument.

Btw, the code using this method ignores the return value entirely, which is a clear indication that there is no meaningful one.

and regarding the fluent interface, I suggest you to read http://ocramius.github.io/blog/fluent-interfaces-are-evil/ as there are good arguments in it